### PR TITLE
add log,db,exception framwork

### PR DESCRIPTION
**Pull request by xxwjj**
add log, db, exception framwork.

### CODE REVIEW

File: anomaly_detection/db/models.py [hunk 1]
```diff
@@ -13,13 +13,162 @@
...
+class TimestampMixin(object):
+    created_at = Column(DateTime, default=lambda:  datetime.datetime.utcnow())
+    updated_at = Column(DateTime, onupdate=lambda: datetime.datetime.utcnow())
+
+
+Base = declarative_base()
+
+
+class AnomalyDetectionBase(ModelBase, TimestampMixin):
+    """Base class for Manila Models."""
```
Comment (xing-yang) [inline anomaly_detection/db/models.py]: Manila?  Copy and paste error

File: anomaly_detection/db/models.py [hunk 2]
```diff
@@ -13,13 +13,162 @@
...
+    __table_args__ = {'mysql_engine': 'InnoDB'}
```
Comment (xing-yang) [inline anomaly_detection/db/models.py]: We should use mongoDB, not mysql.

Also we should make the database a pluggable framework to allow different database type to be used in the future.

File: anomaly_detection/cmd/api.py [hunk 3]
```diff
@@ -25,7 +26,7 @@ def __init__(self):
         self._init_server()
 
     def _init_logging(self):
-        pass
+        log.setup(log.Config, "manila")
```
Comment (xing-yang) [inline anomaly_detection/cmd/api.py]: Not "manila".
Please remove "manila" in all places.

**Comment by xing-yang** [comment]
Hi @xxwjj , we should use MongoDB, not mysql.

Also please make the database layer pluggable to allow different types of databases to be used by 3rd Party (3rd Party may want to use our ML service but use a different type of database).

File: anomaly_detection/db/api.py [hunk 4]
```diff
@@ -11,211 +11,99 @@
...
+                   lazy=lazy)
+
+# return cls(backend_name=conf.database.backend,
+#            backend_mapping=backend_mapping,
+#            lazy=lazy,
+#            use_db_reconnect=conf.database.use_db_reconnect,
+#            retry_interval=conf.database.db_retry_interval,
+#            inc_retry_interval=conf.database.db_inc_retry_interval,
+#            max_retry_interval=conf.database.db_max_retry_interval,
+#            max_retries=conf.database.db_max_retries)
```
Comment (xing-yang) [inline anomaly_detection/db/api.py]: Remove commented out lines.

File: anomaly_detection/db/api.py [hunk 5]
```diff
@@ -11,211 +11,99 @@
...
-    engine = get_engine()
-    models.Base.metadata.create_all(engine)
+from anomaly_detection.utils import uuid
+from anomaly_detection import context as ctx
 
 
 if __name__ == '__main__':
-    init_db()
-
+    # __import__("anomaly_detection.db.sqlalchemy")
```
Comment (xing-yang) [inline anomaly_detection/db/api.py]: Remove commented out line.

File: anomaly_detection/utils/__init__.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2012 Intel Corporation.
```
Comment (xing-yang) [inline anomaly_detection/utils/__init__.py]: 2019 OpenSDS Authors

File: anomaly_detection/cmd/mange.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 The OpenSDS Authors.
```
Comment (xing-yang) [inline anomaly_detection/cmd/mange.py]: s/mange.py/manage.py

File: anomaly_detection/cmd/mange.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+            parser.set_defaults(action_kwargs=action_kwargs)
+
+
+def main():
+
+    script_name = sys.argv[0]
+    if len(sys.argv) < 2:
+        print(script_name + " category action [<args>]")
+        print("Available categories:")
+        for category in CATEGORIES:
```
Comment (xing-yang) [inline anomaly_detection/cmd/mange.py]: What are available categories?

File: anomaly_detection/db/api.py [hunk 9]
```diff
@@ -11,211 +11,99 @@
...
+#            max_retries=conf.database.db_max_retries)
+
+
+class Config:
+    class Database:
+        backend = "sqlalchemy"
+    database = Database
+
+
+_BACKEND_MAPPING = {'sqlalchemy': 'anomaly_detection.db.sqlalchemy.api'}
```
Comment (xing-yang) [inline anomaly_detection/db/api.py:69]: Add "TODO: Add support for other types of databases in a plugin model".

File: anomaly_detection/cmd/mange.py (hunk 8 above)
Comment (xxwjj) [reply anomaly_detection/cmd/mange.py]: It's only support db category right now.

File: anomaly_detection/db/models.py (hunk 2 above)
Comment (xxwjj) [reply anomaly_detection/db/models.py]: done

File: anomaly_detection/cmd/api.py (hunk 3 above)
Comment (xxwjj) [reply anomaly_detection/cmd/api.py]: done

File: anomaly_detection/db/api.py (hunk 4 above)
Comment (xxwjj) [reply anomaly_detection/db/api.py]: done

File: anomaly_detection/db/api.py (hunk 5 above)
Comment (xxwjj) [reply anomaly_detection/db/api.py]: done

**Review by xing-yang — APPROVED** [review]
lgtm

**Pull request merged**